cli/azd/pkg/prompt/prompter.go

```diff
@@ -218,6 +219,10 @@ func (p *DefaultPrompter) getSubscriptionOptions(ctx context.Context) ([]string,
 		return nil, nil, nil, fmt.Errorf("listing accounts: %w", err)
 	}
 
+	sort.Slice(subscriptionInfos, func(i, j int) bool {
+		return strings.ToLower(subscriptionInfos[i].Name) < strings.ToLower(subscriptionInfos[j].Name)
```

Comment:
The calls to `strings.ToLower`  here ends up allocating 2x string copies for each element.

A way to avoid string allocations is to leverage slices and lowercasing each character rune. This kind of string comparison doesn't work for all languages -- but it would work for ASCII usage.